### PR TITLE
mid data defined

### PR DESCRIPTION
**Pull request by javibu13**

### CODE REVIEW

**Review by mariuscrsn — APPROVED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Review by mariuscrsn — DISMISSED** [review]

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+
+#######################             CLASSES              #######################
+
+class MidDataBatteryC:
+    '''
+    Generic battery info.
+    '''
+    # pylint: disable=too-many-arguments
+    def __init__(self, name: str, model : str, volt_min : int = 800,
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py]: All params must be initialized to None

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_common.py [hunk 2]
```diff
@@ -4,9 +4,8 @@
 '''
 #######################        MANDATORY IMPORTS         #######################
 from __future__ import annotations
-
+from copy import deepcopy
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_common.py]: Remove deepcopy

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_common.py (hunk 2 above)
Comment (RobertoAldea) [reply code/src/wattrex_battery_cycler/mid/mid_data/mid_data_common.py]: Done

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py (hunk 1 above)
Comment (RobertoAldea) [reply code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py]: Done, also Lithium Bat and Redox Bat initialize to None

Comment (RobertoAldea) [reply code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py]: Should I initilize MidDataGenericMeas also to None or with value 0? @mariuscrsn

Comment (mariuscrsn) [reply code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py]: To 0

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+log = sys_log_logger_get_module_logger(__name__)
+#######################          PROJECT IMPORTS         #######################
+
+#######################          MODULE IMPORTS          #######################
+from .mid_data_experiment import MidDataPwrRangeC
+#######################              ENUMS               #######################
+
+#######################             CLASSES              #######################
+
+class MidDataBatteryC:
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py:23]: Use MidDataPwrRange

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    CURRENT = 2
+    POWER   = 3
+
+
+#######################             CLASSES              #######################
+class MidDataCyclerStationC:
+    '''
+    Cycler station information.
+    '''
+    def __init__(self, cycler_id: int, name: str, devices: List[MidDataDeviceC],
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py]: Initialize to None and `[]`

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        self.name : str = name
+        self.cycler_id : int = cycler_id
+        self.devices : List[MidDataDeviceC] = devices
+        self.deprecated: bool = deprecated
+
+class MidDataInstructionC:
+    '''
+    Instruction to applied on experiments.
+    '''
+    def __init__(self, instr_id: int, mode : MidDataPwrModeE, ref : int,
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py]: Initialize to None

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+        self.ref : int = ref
+        self.limit_type : MidDataPwrLimitE = limit_type
+        self.limit_ref : int = limit_ref
+
+
+class MidDataExperimentC:
+    '''
+    Experiment relate information.
+    '''
+    def __init__(self, exp_id : int, name : str, date_begin : datetime,
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py]: Initialize to None

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+        self.name : str = name
+        self.date_begin : datetime = date_begin
+        self.date_finish : datetime = date_finish
+        self.status : MidDataExpStatusE = status
+
+class MidDataPwrRangeC:
+    '''
+    Power range that specify operational limits for voltage and current.
+    '''
+    def __init__(self, volt_max : int, volt_min : int, curr_max : int,
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py]: Initialize to None

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        self.volt_min : int = volt_min
+        self.curr_max : int = curr_max
+        self.curr_min : int = curr_min
+
+
+class MidDataProfileC:
+    '''
+    The class of MidDataProfileC is a class that is used for the MIDDataProfileC .
+    '''
+    def __init__(self, name : str, power_range : MidDataPwrRangeC,
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py]: Initialize to None

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+        self.instructions : List[MidDataInstructionC] = instructions
+        self.range : MidDataPwrRangeC = power_range
+
+
+class MidDataAlarmC:
+    """
+    Alarm triggered to notify that an unexpected event has occurred
+    during cycler operation.
+    """
+    def __init__(self, timestamp : datetime, code : int, value : int):
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_experiment.py]: Initialize to None

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_devices.py [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+            - None
+        '''
+        return self.__status.name
+
+class MidDataLinkConfSerialC: #pylint: disable=too-many-instance-attributes
+    '''
+    A class method that implements the MIDDataLinkConSerialf class .
+    '''
+    # pylint: disable=too-many-arguments
+    def __init__(self, separator: str, baudrate: int, bytesize: int,
```
Comment (mariuscrsn) [inline code/src/wattrex_battery_cycler/mid/mid_data/mid_data_devices.py]: Remove arguments and use kwargs

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py (hunk 1 above)
Comment (mariuscrsn) [reply code/src/wattrex_battery_cycler/mid/mid_data/mid_data_battery.py]: Why these numbers are used?

**Review by mariuscrsn — APPROVED** [review]

**Review by mariuscrsn — APPROVED** [review]

**Pull request merged**